src/primitives/common/linear_equation.rs

```diff
@@ -3,62 +3,194 @@ use crate::{
     primitives::{common::LineSide, Line},
 };
 
-/// Linear equation representation
+/// Scaling factor for unit length normal vectors.
+const NORMAL_VECTOR_SCALE: i32 = 1 << 10;
+
+/// Linear equation.
 ///
-/// The equation is stored as the a, b and c coefficients of the ax + by + c = 0 equation
+/// The equation is stored as a normal vector and the distance to the origin.
 #[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
-pub struct LinearEquation<T> {
-    pub a: T,
-    pub b: T,
-    pub c: T,
+pub struct LinearEquation {
+    /// Normal vector, perpendicular to the line.
+    ///
+    /// The unit vector is scaled up to increase the resolution.
+    pub normal_vector: Point,
+
+    /// Distance from the origin.
+    ///
+    /// The distance doesn't directly correlate to the distance in pixels, but is
+    /// scaled up by the length of the normal vector.
+    pub origin_distance: i32,
 }
 
-impl LinearEquation<i32> {
-    pub const fn from_line(line: &Line) -> Self {
+impl LinearEquation {
+    /// Creates a new linear equation from a line.
+    pub fn from_line(line: &Line) -> Self {
+        let normal_vector = line.delta().rotate_90();
+        let origin_distance = line.start.dot_product(normal_vector);
+
         Self {
-            a: line.end.y - line.start.y,
-            b: line.start.x - line.end.x,
-            c: line.end.x * line.start.y - line.start.x * line.end.y,
+            normal_vector,
+            origin_distance,
         }
     }
-}
 
-impl LinearEquation<Real> {
-    /// Create a new linear equation based on one point and one angle
+    /// Creates a new linear equation based on one point and one angle.
     pub fn from_point_angle(point: Point, angle: Angle) -> Self {
         // FIXME: angle.tan() for 180.0 degrees isn't exactly 0 which causes problems when drawing
```

Comment:
I assume this comment is still valid and will get fixed later, right?

Reply:
It is still valid and it might get fixed later.